leo/src/DAO/supabase/SupabaseEventOccurrenceDAO.ts

```diff
@@ -0,0 +1,70 @@
+import { EventOccurrenceDAO } from "@/DAO/interface/EventOccurrenceDAO";
+import { createClient } from "@/utils/supabase/server";
+import { Tables, TablesInsert, TablesUpdate } from "../../../types/database.types";
+
+export class SupabaseEventOccurrenceDAO implements EventOccurrenceDAO {
+    private supabase = createClient();
+    private TABLE = 'Event_Occurrences'
+
+    async getEventOccurrences(): Promise<Tables<'Event_Occurrences'>[]> {
+        const { data, error } = await this.supabase.from(this.TABLE).select()
+        if (error) { throw error }
+        return data ?? []
```

Comment:
how can data be undefined here? wouldnt that just be an error?

Reply:
If you look at the return type of data then it can be an array or null